### DISCUSSION TITLE
rule jsonPath to select properties names containing '@' symbol

### DISCUSSION THREAD

**Discussion by asatarov** [stoplightio/spectral#1678] [Q&A]
I’m trying to create rule for model properties with ‘@’ symbols in name. I use 
given: $.definitions.*.properties[?(@property.match(/@/))]~

but it doesn’t work. It works for some other regexps but not for all. For me it is not clear how it works. And may be ‘@’ symbol has some special meaning here.

Could someone give me some hints how to use it in a right way and is it possible to debug it somehow?

And I use https://extendsclass.com/jsonpath-tester.html for testing.
And may be there is a better tool for test json path ?

Example spec:

```json
{
   "swagger": "2.0",
   "info": {
      "title": "Product API",
      "version": "1.0.0",
      "description": "description"
   },
   "paths": {
      "/PRODUCTS": {
         "get": {
            "summary": "Query Product Catalogue",
            "tags": [],
            "responses": {},
            "operationId": "QUERY-PRODUCTS",
            "description": ""
         }
      },
      "/Products/TYPE": {
         "get": {
            "summary": "Query Product byb type",
            "tags": [],
            "responses": {},
            "operationId": "get-Products-TYPE"
         }
      }
   },
   "definitions": {
      "product": {
         "title": "product",
         "type": "object",
         "properties": {
            "@productId": {
               "type": "string"
            },
            "@baseType": {
               "type": "string"
            },
            "otherPropery": {
               "type": "string"
            }
         }
      }
   }
}

```

**Comment by P0lip** [comment]
@asatarov 
Hey!
Try the following rule
```yaml
  rules:
    test:
      given: $.definitions[*].properties[?(@property ~= "@")]
      severity: error
      then:
        field: "@key"
        function: falsy
```

> And may be there is a better tool for test json path ?

Spectral uses [nimma](https://github.com/p0lip/nimma) under the hood for a JSON Path engine.
There's a REPL available [here](https://nimma.rozek.tech/)
